controllers/nstemplatetier/nstemplatetier_controller_test.go

```diff
@@ -395,6 +395,56 @@ func TestUpdateNSTemplateTier(t *testing.T) {
 				})
 		})
 	})
+
+	t.Run("when updating one tiertemplate the revisions field should be cleaned up from old entries", func(t *testing.T) {
+		// given
+		tierBeingUpdated := oldNSTemplateTier.DeepCopy()
+		tierBeingUpdated.Name = "base1nsupdate"
+		// we update the reference of one tierTemplate
+		clusterResourceTierTemplate := createTierTemplate(t, "clusterresources", withTemplateObjects(crq), tierBeingUpdated.Name, "123456updated")
+		tierBeingUpdated.Spec.ClusterResources.TemplateRef = clusterResourceTierTemplate.GetName()
+		r, req, cl := prepareReconcile(t, tierBeingUpdated.Name, append(tierTemplates, tierBeingUpdated, clusterResourceTierTemplate)...)
+
+		// when
+		res, err = r.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+		// revisions values should be different compared to the previous ones
+		// ensure the old revisions are not there anymore, and instead the update one is in the list
+		tierTemplatesRefs := []string{
+			"base1ns-admin-123456new", "base1nsupdate-clusterresources-123456updated", "base1ns-code-123456new", "base1ns-dev-123456new", "base1ns-edit-123456new", "base1ns-stage-123456new", "base1ns-viewer-123456new",
+		}
+		newNSTmplTier := tiertest.AssertThatNSTemplateTier(t, tierBeingUpdated.Name, cl).
+			HasStatusTierTemplateRevisions(tierTemplatesRefs).Tier()
```

Comment:
btw, you need to update also the `HasStatusTierTemplateRevisions` so it checks that the length is the same. Currently, it only checks that the refs are present, but it doesn't catch the case when there are also other refs that shouldn't be there.

Reply:
you're right! Great catch! Thanks! 

Fixed in https://github.com/codeready-toolchain/host-operator/pull/1136/commits/a9916b1164f67999b1ae5278957d72b9be086148